scripts/test_images.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+# Exit on error
+set -e
+
+# Consider command as failed when any component of the pipe fails:
+# https://stackoverflow.com/questions/1221833/pipe-output-and-capture-exit-status-in-bash
+set -o pipefail
+
+source ./scripts/test_lib.sh
+source ./scripts/build_lib.sh
+
+function startContainer {
+    # run docker in the background
+    docker run -d --rm --name "${RUN_NAME}" "${IMAGE}" &
+
+    # wait for etcd daemon to bootstrap
+    sleep 5
+}
+
+function stopContainer {
+    docker stop "${RUN_NAME}"
+    docker image rm "${IMAGE}"
+}
+
+# Can't proceed without docker
+if ! command -v docker >/dev/null; then
+    log_error "cannot find docker"
+    exit 1
+fi
+
+# You can't run darwin binaries in linux containers
+if [[ $(go env GOOS) == "darwin" ]]; then
+    echo "Please use linux machine for release builds."
+    exit 1
+fi
+
+# Docker build
+ARCH=$(go env GOARCH)
+DOCKERFILE="Dockerfile-release.${ARCH}"
+BINARYDIR=${BINARYDIR:-"bin"}
+# Pick defaults based on release workflow
+REPOSITARY=${REPOSITARY:-"gcr.io/etcd-development/etcd"}
+if [ -n "$VERSION" ]; then
+    # Expected Format: v3.6.99-amd64
+    TAG=v"${VERSION}"-"${ARCH}"
+else
+    echo "Terminating test, VERSION not supplied"
+    exit 1
+fi
+IMAGE=${IMAGE:-"${REPOSITARY}:${TAG}"}
+
+# ETCD related values
+RUN_NAME="test_etcd"
+KEY="foo"
+VALUE="bar"
+
+# Build if binaries are not present
+if [ -z "$BINARYDIR" ]; then
+    run ./scripts/build.sh
+fi
+
+# Build only if image is not present
+if [[ "$(docker images -q "${IMAGE}" 2> /dev/null)" == "" ]]; then
+    echo "${TAG} not present locally, building it..."
+    # Build a local image from bin directory
+    if ! docker build -t "${IMAGE}" -f "${DOCKERFILE}" "${BINARYDIR}"; then
+        echo "Docker build unsuccessful. Exit code $?"
+        exit 1
+    fi
+fi
```

Comment:
It seems that the $BINARYDIR doesn't exist even the image is available.

```suggestion
# Build only if image is not present
if [[ "$(docker images -q "${IMAGE}" 2> /dev/null)" == "" ]]; then
    echo "${TAG} not present locally, building it..."
    
    # Build if binaries are not present
    if [ -z "$BINARYDIR" ]; then
        run ./scripts/build.sh
    fi
    
    # Build a local image from bin directory
    if ! docker build -t "${IMAGE}" -f "${DOCKERFILE}" "${BINARYDIR}"; then
        echo "Docker build unsuccessful. Exit code $?"
        exit 1
    fi
fi
```

I am thinking to totally remove the above build script for simplicity, we can assume the image is already available (fail the test if not present).

Reply:
I initially created the script so that it can work in a stand-alone manner, this is what is followed in other scripts like build-docker.sh [1], so my vote is to keep the above changes. 


[1]
scripts/build-docker.sh

> \#If binarydirectory is not specified download tarballs and extract...
> if [ -z "${BINARYDIR}" ]; then
>   RELEASE="etcd-${1}"-$(go env GOOS)-$(go env GOARCH)
>   BINARYDIR="${RELEASE}"
>   TARFILE="${RELEASE}.tar.gz"
>   TARURL="https://github.com/etcd-io/etcd/releases/download/${1}/${TARFILE}"
>   if ! curl -f -L -o "${TARFILE}" "${TARURL}" ; then
>     echo "Failed to download ${TARURL}."
>     exit 1
>   fi
>   tar -zvxf "${TARFILE}"
> fi